app/src/main/java/com/example/cr554/newsapp/NewsViewHolder.java

```diff
@@ -37,6 +39,8 @@ public void bindNews(NewsArticle news){
     }
 
     public void onClick(View view){
-        //opens a webview of the article
+        NewsArticle newsArticle = MainActivity.newsArticles.get(getAdapterPosition());
+        Intent browserIntent = new Intent("android.intent.action.VIEW", Uri.parse(newsArticle.getUrl()));
```

Comment:
Oh, I see. I misread -- if you would like to get the address of a specific story, the API should be returning the correct address for you at the time of the query (before you even display the item in the list of articles). In that case, you can always attach that information to the item in the view via the adapter. Does that make sense? :)

The above that you proposed works just as well! :)